File: projects/engineering.purdue.edu/scriptingwithobjects/swocode/chap4/Anchor.pl
#!/usr/bin/perl -w                                          
  
## Anchor.pl

use strict;                                                           

my $pattern = '^abra';                                              #(A)

my $input_string = "abracadabra";                                   #(B)
$input_string  =~  /$pattern/ ? print "Yes\n"          # Yes        #(C)
                              : print "No\n";                       #(D)

$input_string = "dabracababra";                                     #(E)
$input_string  =~  /$pattern/ ? print "Yes\n"                       #(F)
                              : print "No\n";          # No         #(G)

$pattern = 'dabra$';                                                #(H)

$input_string = "abracadabra";                                      #(I)
$input_string  =~  /$pattern/ ? print "Yes\n"          # Yes        #(J)
                              : print "No\n";                       #(K)

$input_string = "dabracababra";                                     #(L)
$input_string  =~  /$pattern/ ? print "Yes\n"                       #(M)
                              : print "No\n";          # No         #(N)

$pattern = '\bwhat';                                                #(O)

$input_string = "whatever will be will be";                         #(P)
$input_string  =~  /$pattern/ ? print "Yes\n"          # Yes        #(Q)
                              : print "No\n";                       #(R)

$input_string = "somewhat happier than thou";                       #(S)
$input_string  =~  /$pattern/ ? print "Yes\n"                       #(T)
                              : print "No\n";          # No         #(U)

$pattern = 'ever\b';                                                #(V)

$input_string = "whatever will be will be";                         #(W)
$input_string  =~  /$pattern/ ? print "Yes\n"          # Yes        #(X)
                              : print "No\n";                       #(Y)   

$input_string = "everywhere I go you go";                           #(Z)
$input_string  =~  /$pattern/ ? print "Yes\n"                       #(a)
                              : print "No\n";          # No         #(b)

File: projects/demos/engineering.purdue.edu/scriptingwithobjects/swocode/chap4/Anchor.py
#!/usr/bin/python
  
## Anchor.py

import re                                                           #(A)

pattern = r'^abra'                                                  #(B)

input_string = "abracadabra";                                       #(C)
m = re.search( pattern, input_string )                              #(D)
if m != None :                                                     
    print 'Yes'                                 # Yes
else :
    print 'No'

input_string = "dabracababra"                                       #(E)
m = re.search( pattern, input_string )                              #(F)
if m != None :                                                     
    print 'Yes'
else :
    print 'No'                                  # No


pattern = r'dabra$'                                                 #(G)

input_string = "abracadabra"                                        #(H)
m = re.search( pattern, input_string )                              #(I)
if m != None :                                                     
    print 'Yes'                                 # Yes
else :
    print 'No'

input_string = "dabracababra"                                       #(J)
m = re.search( pattern, input_string )                              #(K)
if m != None :                                                     
    print 'Yes'
else :
    print 'No'                                  # No


pattern = r'\bwhat'                                                 #(L)

input_string = "whatever will be will be"                           #(M)
m = re.search( pattern, input_string )                              #(N)
if m != None :                                                    
    print "Yes"                                 # Yes
else :
    print "No"    

input_string = "somewhat happier than thou"                         #(O)
m = re.search( pattern, input_string )                              #(P)
if m != None :
    print "Yes"   
else :
    print "No"                                  # No


pattern = r'ever\b'                                                 #(Q)

input_string = "whatever will be will be"                           #(R)
m = re.search( pattern, input_string )
if m != None :
    print "Yes"                                 # Yes
else :
    print "No"    

input_string = "everywhere I go you go"                             #(S)
m = re.search( pattern, input_string )                              #(T)
if m != None :
    print "Yes" 
else :
    print "No"                                  # No  
